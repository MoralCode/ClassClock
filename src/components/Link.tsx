import React, { Component, CSSProperties } from "react";

interface ILinkProps {
    destination: any;
    className?: string;
    style?: CSSProperties;
    id?: string;
}

export default class Link extends Component<ILinkProps, {}> {
    render() {
        return (
            <a
                className={this.props.className}
                id={this.props.id}
                style={this.props.style}
                href={
                    typeof this.props.destination === "function"
                        ? undefined
                        : this.props.destination
                }
                onClick={
                    typeof this.props.destination === "function"
                        ? this.props.destination
                        : undefined
                }
            >
                {this.props.children}
            </a>
        );
    }
}
