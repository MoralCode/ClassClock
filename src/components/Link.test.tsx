import React from "react";
import renderer from "react-test-renderer";
import Link from "./Link";

test("Link renders correctly with a static link", () => {
    const component = renderer.create(
        <Link destination="https://classclock.app" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // // manually trigger the callback
    // tree.props.onMouseEnter();
    // re-rendering
    // tree = component.toJSON();
    // expect(tree).toMatchSnapshot();

    // manually trigger the callback
    // tree.props.onMouseLeave();
    // re-rendering
    // tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
});

it("renders correctly with a function", () => {
    let pass = false;
    const onClick = () => {pass = true}
    const component = renderer.create(<Link destination={onClick} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    if (tree != null) {
        // manually trigger the callback
        tree.props.onClick();
    }
    

    expect(pass).toBeTruthy();
    
});
