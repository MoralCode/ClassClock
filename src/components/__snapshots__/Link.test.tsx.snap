// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component renders correctly with a function 1`] = `
<a
  onClick={[Function]}
/>
`;

exports[`Link Component renders correctly with a static link 1`] = `
<a
  href="https://classclock.app"
/>
`;
